File: dotnet/Vaas/src/Vaas/VaasOptions.cs
using System;
using System.ComponentModel.DataAnnotations;
using Vaas.Authentication;

namespace Vaas;

public class VaasOptions
{
    public Uri Url { get; set; } = new("https://upload.production.vaas.gdatasecurity.de");
    public bool? UseHashLookup { get; init; } = null;
    public bool? UseCache { get; init; } = null;

    public Uri TokenUrl { get; set; } =
        new Uri("https://account.gdata.de/realms/vaas-production/protocol/openid-connect/token");

    [Required]
    [CustomValidation(typeof(TokenRequest), nameof(TokenRequest.IsValid))]
    public TokenRequest Credentials { get; set; } = null!;
}

File: php/src/vaas/Http/Options/VaasOptions.php
<?php

namespace VaasSdk\Options;

use VaasSdk\Authentication\TokenRequest;

class VaasOptions
{
    public TokenRequest $credentials;
    public ?bool $useHashLookup;
    public ?bool $useCache;
    public string $url;
    public string $tokenUrl;

    public function __construct(
        TokenRequest $credentials,
        ?bool $useHashLookup = null,
        ?bool $useCache = null,
        string $url = 'https://gateway.production.vaas.gdatasecurity.de/',
        string $tokenUrl = 'https://account.gdata.de/realms/vaas-production/protocol/openid-connect/token'
    ) {
        $this->url = $url;
        $this->useHashLookup = $useHashLookup;
        $this->useCache = $useCache;
        $this->tokenUrl = $tokenUrl;
        $this->credentials = $credentials;
    }
}